netty/shaded/build.gradle

```diff
@@ -45,21 +45,18 @@ shadowJar {
 publishing {
     publications {
         maven(MavenPublication) {
-            // Ideally swap to project.shadow.component(it) when it isn't broken for project deps
-            artifact shadowJar
+            project.shadow.component(it)
+
             // Empty jars are not published via withJavadocJar() and withSourcesJar()
             artifact javadocJar
             artifact sourcesJar
 
+            // shadow.component() is run after the main build.gradle's withXml
             pom.withXml {
-                def dependencies = asNode().appendNode('dependencies')
-                project.configurations.shadow.allDependencies.each { dep ->
-                    def dependencyNode = dependencies.appendNode('dependency')
-                    dependencyNode.appendNode('groupId', dep.group)
-                    dependencyNode.appendNode('artifactId', dep.name)
-                    def version = (dep.name == 'grpc-core') ? '[' + dep.version + ']' : dep.version
-                    dependencyNode.appendNode('version', version)
-                    dependencyNode.appendNode('scope', 'compile')
+                asNode().dependencies.'*'.findAll() { dep ->
+                    dep.artifactId.text() in ['grpc-api', 'grpc-core']
```

Comment:
Should 'grpc-context' be included as well?

Reply:
No. These are present because they contain Internal (and experimental) APIs. grpc-context exposes no internal APIs used by other grpc modules.

Reply:
Might be worth a comment here.

Reply:
I thought that is what the "shadow.component() is run after the main build.gradle's withXml" was. I'm not trying to explain what is here, I want to just say, "re-do whatever the main build.gradle would have done." Do you want me to split it out into a function?

Reply:
Oh, we do this in main build already. Not just for shading. That's fine as-is.